src/main/java/com/flansmod/common/guns/FireableGun.java

```diff
@@ -0,0 +1,85 @@
+package com.flansmod.common.guns;
+
+import com.flansmod.common.types.InfoType;
+
+/**
+ * Class used for storing the properties of a gun
+ */
+public class FireableGun
```

Comment:
GunType is a specific data storage for guns which exists as items. This makes it impossible to use GunType as general data storage because for example AAGunType is completely different class which cant be converted into GunType.
FireableGun unifys all the important data about the gun, making the code using FireableGun very simple because the acquiring of the data is done by the code which calls the fireGun method